src/components/checkbox/checkbox.tsx

```diff
@@ -0,0 +1,41 @@
+import React = require('react');
+
+interface CheckBoxProps {
+    text?: string;
+    value?: boolean;
+    boxIcon?: string;
+    tickIcon?: string;
+    onChange?: (value: boolean) => any;
+
+}
+
+export const CheckBox: React.SFC<CheckBoxProps> = (props) => (
+    <div data-automation-id="CHECKBOX_ROOT"
+         onClick={(event) => executeClickHandler(props.onChange!, !props.value)}>
+        <img data-automation-id="CHECKBOX_BOX"
+             src={props.boxIcon}
+             style={{width: '1em', height: '1em'}}/>
+        <img data-automation-id="CHECKBOX_TICKMARK"
+             src={props.tickIcon}
+             style={{
+                 width: '1em',
+                 height: '1em',
+                 visibility: props.value ? 'visible' : 'hidden',
+                 position: 'relative',
+                 marginLeft: '-1em'
+             }}/>
+        <span data-automation-id="CHECKBOX_LABEL">{props.text || ''}</span>
+    </div>
+);
+
+function executeClickHandler(handler: (value: boolean) => any, value: boolean) {
+    handler && handler(value);
```

Comment:
I actually believe that every event handler property should have nop as default value. The reason is that the component shouldn't break when used naively, without any explicit props. Otherwise we would have to check before every invocation of every handler.